pkg/gits/bitbucket_server.go

```diff
@@ -725,6 +745,21 @@ func (b *BitbucketServerProvider) MergePullRequest(pr *GitPullRequest, message s
 func (b *BitbucketServerProvider) CreateWebHook(data *GitWebHookArguments) error {
 	projectKey, repo := parseBitBucketServerURL(data.Repo.URL)
 
+	if data.URL == "" {
+		return errors.New("missing property URL")
+	}
+
+	hooks, err := b.ListWebHooks(projectKey, repo)
+	if err != nil {
+		log.Errorf("Error querying webhooks on %s/%s: %s\n", projectKey, repo, err)
```

Comment:
Shouldn't it return an error here? If this fails, the `hooks` slice is empty, and then the for will not execute and it will return the hook with the data.URL, even thought maybe there is already a weebook registered with the same URL but it couldn't be listed.